demux/demux_mf.c

```diff
@@ -119,11 +119,12 @@ static mf_t *open_mf_pattern(void *talloc_ctx, struct demuxer *d, char *filename
         goto exit_mf;
     }
 
-    size_t fname_avail = strlen(filename) + 32;
+    bstr bfilename = bstr0(filename);
+    size_t fname_avail = bfilename.len + 32;
     char *fname = talloc_size(mf, fname_avail);
 
 #if HAVE_GLOB
-    if (!strchr(filename, '%')) {
+    if (!mp_is_url(bfilename) && !strchr(filename, '%')) {
```

Comment:
considering none of this demuxer is made for use with URLs shouldn't they be rejected right at the start?

otoh it's not our problem if the libc has bugs